version: '3.8'

services:
  traefik:
    image: traefik:v2.10
    container_name: traefik_proxy
    restart: always
    ports:
      - "80:80"
    command:
      - "--api.insecure=true"
      - "--providers.docker=true"
      - "--entrypoints.web.address=:80"
    volumes:
      - "/var/run/docker.sock:/var/run/docker.sock"

  web:
    build: .
    image: django-app:latest
    container_name: django_app
    restart: always
    env_file:
      - .env
    labels:
      - "traefik.enable=true"
      - "traefik.http.routers.django.rule=Host(`13.125.246.239`)"  
      - "traefik.http.services.django.loadbalancer.server.port=8000"

  celery_worker:
    build: .
    image: django-app:latest
    container_name: celery_worker
    restart: always
    depends_on:
      - web
    env_file:
      - .env
    command: celery -A ticketing_project worker --loglevel=info